src/main/software/amazon/event/ruler/ByteMachine.java

```diff
@@ -1680,16 +1684,15 @@ private void addMatchReferences(ByteMatch match) {
                 hasNumeric.incrementAndGet();
             }
             break;
+        case ANYTHING_BUT_PREFIX:
         case ANYTHING_BUT_IGNORE_CASE:
+        case ANYTHING_BUT_WILDCARD:
             addToAnythingButsMap(anythingButs, match.getNextNameState(), match.getPattern());
             break;
         case ANYTHING_BUT_SUFFIX:
             hasSuffix.incrementAndGet();
```

Comment:
A lot of the special logic around hasNumeric / hasSuffix can be abstracted away if there was method here that recursively called the same switch function. It an overkill to be combined with this change, but a follow-up change would help make anything-but way more compatible with all future matchers (and remove the ANYTHING_BUT_<MATCHER> constants.